src/phoenix/experimental/evals/functions/binary.py

```diff
@@ -47,18 +48,22 @@ def llm_eval_binary(
 
         system_instruction (Optional[str], optional): An optional system message.
 
+        verbose (bool, optional): If True, prints detailed info to stdout. Default False.
+
     Returns:
         List[str]: A list of strings representing the predicted class for each record in the
         dataframe. The list should have the same length as the input dataframe and its values should
         be the entries in the rails argument or "NOT_PARSABLE" if the model's prediction could not
         be parsed.
     """
 
-    eval_template = normalize_template(template)
-    prompts = map_template(dataframe, eval_template)
-    responses = model.generate(prompts.to_list(), instruction=system_instruction)
-    rails_set = set(rails)
-    return [_snap_to_rail(response, rails_set) for response in responses]
+    with set_verbosity(model, verbose) as m:
```

Comment:
Wouldn't it be simpler to just leverage the fact that `model` is a `dataclasse`, and wouldn't making a copy be better than mutating state?
```suggestion
    model = replace(model, _verbose=verbose)
```

Reply:
@RogerHYang I think that's great too, but practically I find the two approaches essentially interchangeable

Reply:
@axiomofjoy addressed by [9f863e6](https://github.com/Arize-ai/phoenix/pull/1558/commits/9f863e6bf86523cd855beb2ce05c56feb5b18cce) to use a better dummy variable name

Reply:
i think mutating state plus context manager seems more complicated than necessary compared to just making a copy of the model

Reply:
at face value it's also not clear that state is being mutated by the context manager

Reply:
i don't think this is ideal

Reply:
@RogerHYang I understand your point there but I think context managers and copies are equally not-complicated, first-class citizens of the Python ecosystem

Reply:
I don't have a problem with context manager. I'm concerned about mutating state when we can just make a copy. We could also make a copy using the context manager, but that would be redundant.

Reply:
The fact that we need a context manager in order to mutate state is an indication that mutating state is more complicated than copying.

Reply:
Both approaches fundamentally require two steps to prevent the persistence of state. In one case we create a copy then change the state. In the other case we change the state then change it back, I really do think both abstractions are quite reasonable ways to approach the problem

Reply:
Yes, they're both reasonable, but I'm recommending the simpler of the two approaches. The number of steps taken is not a factor here, because state mutation is inherently a more dangerous activity than copying.

Reply:
This pattern is specifically recommended by the python logging cookbook here: https://docs.python.org/3/howto/logging-cookbook.html#using-a-context-manager-for-selective-logging

Reply:
And just to clarify, I understand why you prefer another approach, I just want to push back on the suggestion that an extremely common and normal pattern might be considered dangerous

Reply:
No, I don't have a problem with the pattern, because the pattern has to work in many situations. What I am saying is that we have a unique situation in which this pattern can be simplified, i.e. copying instead of mutating state.

Reply:
Below is how I would characterize the decision: I think we should choose approaches that are simpler algorithmically, even though they can be procedurally more complex.
|  | Copy | Mutate |
|---|---|---|
| Algorithmically | simple ✓ | complex |
| Procedurally | complex | simple ✓ |

Reply:
discussed further with @RogerHYang offline—while I think Roger's suggestion is both interesting and a great idea, I think the difference is not a huge deal so I'm going to stick to what we have